### ISSUE TITLE
Enhancement for `reload`

### ISSUE THREAD

**Issue by breadgrocery** [wxt-dev/wxt#1092]
### Feature Request

Currently, the reload function does too little, it only reloads the Inline-Config, nothing more. But the configuration is not all defined in the Inline-Config, they may be configured in different modules. 

It is not ideal to re-run the life cycle when a reload occurs, but the modules are not even aware that a reload has occurred, which leads to the modules not being able to respond to the reload as they should.

```js
  wxt = {
    config,
    hooks,
    hook: hooks.hook.bind(hooks),
    get logger() {
      return config.logger;
    },
    async reloadConfig() {
       wxt.config = await resolveConfig(inlineConfig, command);
    },
    pm,
    builder,
    server
  };
```

For example, the i18n module requires the `default-locale` configuration. However, when reloading, if `default-locale` is not configured through Inline-Config, there will be no `default-locale` configuration in the reloaded configuration, and it will fail in the `prepare:types` phase.

Summary:
- Modules are not aware of reloading and cannot respond to reloading

### What are the alternatives?

My idea is to add a series of hooks about reload (reload:before and others) so that the module can know that a reload is happening and respond correctly when the reload occurs to avoid some problems brought by HMR.
In this way,  no need to rerun the entire life cycle, only the pre-defined functions need to be executed.Something like that
```js
    await reloadConfig();
    wxt.hooks.CallHook("reload:before",()=>{});
    ......
     wxt.hooks.CallHook("reload:done",()=>{});
```

**Comment by aklinker1** [comment]
I am planning on completely restarting the entire context when config changes so module setup functions re-run automatically.

> It is not ideal to re-run the whole life cycle when a reload occurs

Seems like the ideal solution to me, did you see something in the code that would make you think this?

**Comment by breadgrocery** [comment]
> Seems like the ideal solution to me, did you see something in the code that would make you think this?

No. I think this because I see that you only do simple things in the reload step, and I think you may be considering performance or other issues. If there is no problem, then that would be best.
